resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceTypeValidator.scala

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.yarn
+
+import scala.collection.mutable
+
+import org.apache.spark.{SparkConf, SparkException}
+
+object ResourceTypeValidator {
+  private val ERROR_PREFIX: String = "Error: "
+
+  def validateResourceTypes(sparkConf: SparkConf): Unit = {
```

Comment:
Can you write a doc comment explaining what this method is doing? It's not immediately clear from the code what is being validated, what is a valid configuration, etc.

The code in this class also looks extremely complicated for what should be a simple validation of config values.

Reply:
Added doc comment for most of the methods and refactored the whole code in this class, please check whether it became more simple to understand like this.